source/common/upstream/load_stats_reporter.cc

```diff
@@ -60,21 +60,23 @@ void LoadStatsReporter::sendLoadStatsRequest() {
     auto& cluster = it->second.get();
     auto* cluster_stats = request_.add_cluster_stats();
     cluster_stats->set_cluster_name(cluster_name);
-    for (auto& hosts : cluster.hostsPerLocality()) {
-      auto* locality_stats = cluster_stats->add_upstream_locality_stats();
-      ASSERT(hosts.size() > 0);
-      locality_stats->mutable_locality()->MergeFrom(hosts[0]->locality());
-      uint64_t rq_success = 0;
-      uint64_t rq_error = 0;
-      uint64_t rq_active = 0;
-      for (auto host : hosts) {
-        rq_success += host->stats().rq_success_.latch();
-        rq_error += host->stats().rq_error_.latch();
-        rq_active += host->stats().rq_active_.value();
+    for (auto& host_set : cluster.prioritySet().hostSetsPerPriority()) {
```

Comment:
I think we should be merging across all priorities for a given locality here.

Reply:
Really?  I thought we should report stats per-priority (and I have a TODO for that somewhere).
If you prefer merging comment and I'll remove the TODO and merge.

Reply:
As discussed IRL, it would provide greatest flexibility to do it per-priority as you plan. We should do something in the interim to avoid surprises here, since the code as implemented is just going to provide multiple reports when priorities are used for the same locality, with no way to distinguish them (other than implied ordering). I'd suggest merging for now and adding priorities later.

Reply:
We could do that.  Alternately if you don't object I can just only report load for priority group 0, and TODO to add the stats reporting before I fix the load balancers.

Without the load balancers priority-aware it's impossible to get load at other priority clusters, and that's a faster workaround until we do the API change and real load reporting (which is trivial)

Reply:
Yep, priority 0 only and a TODO is fine for this PR, thanks.